File: leetcode/cpp/507_Perfect_Number.cpp
class Solution {
public:
	bool checkPerfectNumber(int num) {
		int sum = 0;
		for (int i = 1; i * i < num; ++i) {
			int into = num / i;
			if (num % i == 0) {
                if (into == num) into = 0;
				sum += (i + into);	
			}
		}
		return sum == num;
	}
};

File: leetcode/cpp/507_Perfect_Number.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num == 1: return False
        total = 0

        for i in range(1, int(num ** 0.5) + 1):
            if num % i == 0:
                pair = num // i

                if pair == num:
                    pair = 0

                total += i + pair

        return total == num
